backend/src/lib.rs

```diff
@@ -24,6 +24,11 @@ pub enum BackendType {
     PilStarkCli,
 }
 
+pub type BackendOptions = String;
```

Comment:
Why enum? Shouldn't it be a struct if anything other than a string? I had a struct at first, but the thing is that each backend has different options. I guess here an enum of different backend options could make sense, but I didn't want to commit to a particular fixed set of options for each backend, the idea being that the backend interface just receives a generic string, and each backend needs to know how to decode it. Otherwise the specific backend information leaks into the user side (eg main or pipeline)

Reply:
I think in both cases, right now, this is essentially specifying an enum variant of a proof type enum (but different enums). I don't understand the variants in the eStark case yet, but I guess my original comment was because I thought the proof type is not really specific to the backend.